src/store/index.jsx

```diff
@@ -1,7 +1,35 @@
-import { configureStore } from '@reduxjs/toolkit';
+import { configureStore } from "@reduxjs/toolkit";
+import { persistReducer, persistStore } from 'redux-persist';
+import { authSlice } from "./slices/auth.slice";
+import { userSlice } from "./slices/user.slice";
+import { homesApi } from './apis/hotels/getPopularHotels.endpoint'; // Importing homesApi
+import { combineReducers } from 'redux';
+import storage from 'redux-persist/lib/storage';
+import {hotelsApi} from './apis/hotels/hotels.api';
 
-import { rootReducer } from './redusers';
+const persistConfig = {
+  key: 'root',
+  storage,
+};
 
+const rootReducer = combineReducers({
+  [authSlice.name]: authSlice.reducer,
+  [userSlice.name]: userSlice.reducer,
+  [hotelsApi.reducerPath]: hotelsApi.reducer,
+  [homesApi.reducerPath]: homesApi.reducer,
+});
+
+const persistedReducer = persistReducer(persistConfig, rootReducer);
+
+// Configure the store
 export const store = configureStore({
-  reducer: rootReducer,
+  reducer: persistedReducer,
+  middleware: (getDefaultMiddleware) =>
+    getDefaultMiddleware({
+      serializableCheck: {
+        ignoredActions: ["persist/PERSIST"],
+      },
+    }).concat(hotelsApi.middleware, homesApi.middleware),
```

Comment:
The same here, `hotelsApi` is enough
```suggestion
    }).concat(hotelsApi.middleware),
```